src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/SpecValidationUtility.cs

```diff
@@ -276,9 +276,9 @@ private static void ValidateStandaloneSpec(PackageSpec spec, IEnumerable<string>
 
         private static void ValidateToolSpec(PackageSpec spec, IEnumerable<string> files)
         {
-            var packageDependencies = GetAllDependencies(spec).ToList();
+            IEnumerable<LibraryDependency> packageDependencies = GetAllDependencies(spec);
 
-            if (packageDependencies.Count != 1
+            if (packageDependencies.Count() != 1
```

Comment:
This is probably going to be O(n). A better approach would be to change the implementation of GetAllDependencies to return an IList, then calling AsList() from here.